src/api/sms/cdac/cdac.service.ts

```diff
@@ -1,82 +1,196 @@
-import { SMS, SMSResponse } from '../sms.interface';
+import {
+  OTPResponse,
+  SMS,
+  SMSData,
+  SMSError,
+  SMSProvider,
+  SMSResponse,
+  SMSResponseStatus,
+  SMSType,
+  TrackResponse,
+} from '../sms.interface';
 
-import { Injectable } from '@nestjs/common';
+import { HttpException, Injectable } from '@nestjs/common';
 import { SmsService } from '../sms.service';
-import fetch from 'node-fetch';
+import { ConfigService } from '@nestjs/config';
+import got, { Got } from 'got/dist/source';
+import * as speakeasy from 'speakeasy';
 
 @Injectable()
 export class CdacService extends SmsService implements SMS {
-  send(id: any): Promise<SMSResponse> {
-    console.log(id);
-    throw new Error('Method not implemented.');
+  baseURL: string;
+  path = '';
+  data: SMSData;
+  httpClient: Got;
+
+  constructor(
+    private configService: ConfigService,
+  ) {
+    super();
+    this.baseURL = configService.get<string>('CDAC_SERVICE_URL');
+    this.httpClient = got;
   }
 
-  track(id: any): Promise<SMSResponse> {
-    console.log(id);
-    throw new Error('Method not implemented.');
+  send(data: SMSData): Promise<SMSResponse> {
+    if (!data) {
+      throw new Error('Data cannot be empty');
+    }
+    this.data = data;
+    if (this.data.type === SMSType.otp) return this.doOTPRequest(data);
+    else return this.doRequest();
+  }
+
+  track(data: SMSData): Promise<SMSResponse> {
+    if (!data) {
+      throw new Error('Data cannot be null');
+    }
+    this.data = data;
+    if (this.data.type === SMSType.otp) return this.verifyOTP(data);
+    else return this.doRequest();
   }
 
-  convertFromUnicodeToText = (message) => {
-    let finalMessage = '';
-    console.log(message);
-    for (let i = 0; i < message.length; i++) {
-      const ch = message.charCodeAt(i);
-      const j = ch;
-      const sss = '&#' + j + ';';
-      finalMessage = finalMessage + sss;
+  private __get_secret(phone): string {
+    return `${this.configService.get<string>('SMS_TOTP_SECRET')}${phone}`
+  }
+
+  private doOTPRequest(data: SMSData): Promise<any> {
```

Comment:
why are we using "__methodname" convention here for private methods in '__get_secret' but not in 'doOTPRequest'

Reply:
No specific reason; but good to keep using the same code formats across the code base. Corrected now.